core/src/main/scala/org/http4s/otel4s/middleware/HttpAttributes.scala

```diff
@@ -0,0 +1,208 @@
+import org.http4s.headers.{Host => _, _}
+import org.typelevel.ci.CIString
+import org.typelevel.otel4s.Attribute
+import org.typelevel.otel4s.semconv.trace.attributes.SemanticAttributes
+
+object HttpAttributes {
+  def clientAddress(
+      forwarded: Option[`X-Forwarded-For`],
+      ip: Option[IpAddress],
+  ): Option[Attribute[String]] = {
+    @inline def fromIp(ip: IpAddress): Attribute[String] =
+      SemanticAttributes.ClientAddress(ip.toUriString)
+    forwarded
+      .map(_.values.toList.flatten.headOption.map(fromIp))
+      .getOrElse(ip.map(fromIp))
+  }
+  def clientPort(port: Port): Attribute[Long] =
+    SemanticAttributes.ClientPort(port.value.toLong)
+  def httpRequestMethod(method: Method): Attribute[String] =
+    SemanticAttributes.HttpRequestMethod(method.name)
+  def httpRequestResendCount(count: Long): Attribute[Long] =
+    SemanticAttributes.HttpRequestResendCount(count)
+  def httpResponseStatusCode(status: Status): Attribute[Long] =
+    SemanticAttributes.HttpResponseStatusCode(status.code.toLong)
+  def networkPeerAddress(ip: IpAddress): Attribute[String] =
+    SemanticAttributes.NetworkPeerAddress(ip.toUriString)
+  def serverAddress(host: headers.Host): Attribute[String] =
+    SemanticAttributes.ServerAddress(headers.Host.headerInstance.value(host))
+  def serverPort(port: Port): Attribute[Long] =
+    SemanticAttributes.ServerPort(port.value.toLong)
+  def urlFull(url: Uri): Attribute[String] =
+    SemanticAttributes.UrlFull(url.renderString)
+  def urlPath(path: Uri.Path): Attribute[String] =
+    SemanticAttributes.UrlPath(path.renderString)
+  def urlQuery(query: Query): Attribute[String] =
+    SemanticAttributes.UrlQuery(query.renderString)
+  def urlScheme(scheme: Uri.Scheme): Attribute[String] =
+    SemanticAttributes.UrlScheme(scheme.value)
+  def userAgentOriginal(userAgent: `User-Agent`): Attribute[String] =
+    SemanticAttributes.UserAgentOriginal(`User-Agent`.headerInstance.value(userAgent))
+
+  private[middleware] def httpRoute(classifiedRoute: String): Attribute[String] =
+    SemanticAttributes.HttpRoute(classifiedRoute)
+
+  object Headers {
+    private def generic(
+        headers: Headers,
+        allowedHeaders: Set[CIString],
+        messageType: String,
+    ): List[Attribute[List[String]]] =
+      headers.headers
+        .groupBy(_.name)
+        .toList
+        .map { case (name, list) =>
+          val key = s"http.$messageType.header.${name.toString.toLowerCase}"
+          if (allowedHeaders.contains(name)) Attribute(key, list.map(_.value))
+          else Attribute(key, List("<REDACTED>"))
+        }
+
+    def request(headers: Headers, allowedHeaders: Set[CIString]): List[Attribute[List[String]]] =
+      generic(headers, allowedHeaders, "request")
+    def response(headers: Headers, allowedHeaders: Set[CIString]): List[Attribute[List[String]]] =
+      generic(headers, allowedHeaders, "response")
+
+    lazy val defaultAllowedHeaders: Set[CIString] = Set(
```

Comment:
Does this correspond to a canonical list somewhere?  If not, alphabetizing them might be easier to maintain in the long run.

Reply:
e🤷‍♀️

Reply:
I've alphebatised them